checkers/regexpPattern_checker.go

```diff
@@ -0,0 +1,68 @@
+package checkers
+
+import (
+	"go/ast"
+	"go/constant"
+	"regexp"
+	"strings"
+
+	"github.com/go-lintpack/lintpack"
+	"github.com/go-lintpack/lintpack/astwalk"
+)
+
+func init() {
+	var info lintpack.CheckerInfo
+	info.Name = "regexpPattern"
+	info.Tags = []string{"diagnostic", "experimental"}
+	info.Summary = "Detects suspicious regexp patterns"
+	info.Before = "regexp.MustCompile(`google.com|yandex.ru`)"
+	info.After = "regexp.MustCompile(`google\\.com|yandex\\.ru`)"
+
+	collection.AddChecker(&info, func(ctx *lintpack.CheckerContext) lintpack.FileWalker {
+		domains := []string{
+			"com",
+			"org",
+			"info",
+			"net",
+			"ru",
+			"de",
+		}
+
+		domainRE := regexp.MustCompile(`[^\\]\.(` + strings.Join(domains, "|") + `)\b`)
+		return astwalk.WalkerForExpr(&regexpPatternChecker{
+			ctx:      ctx,
+			domainRE: domainRE,
+		})
+	})
+}
+
+type regexpPatternChecker struct {
+	astwalk.WalkHandler
+	ctx *lintpack.CheckerContext
+
+	domainRE *regexp.Regexp
+}
+
+func (c *regexpPatternChecker) VisitExpr(x ast.Expr) {
+	call, ok := x.(*ast.CallExpr)
+	if !ok {
+		return
+	}
+
+	switch name := qualifiedName(call.Fun); name {
```

Comment:
```suggestion
	switch qualifiedName(call.Fun) {
```
we don't use name and the func is self-explanatory enough 😉